### PR TITLE
Adjustments to SlowMo to work with recent changes in PyTorch

### PR DESCRIPTION
**Pull request by aovladi**
**What does this PR do? Please describe:**
The very recent change in PyTorch's `torch.distributed.algorithms._comm_hooks` module substituted `AllreduceState` with 
the `DefaultState`. This PR introduces all necessary compatibility changes into SlowMo.

Plus some minor changes in comments and checks.

Testing plan:
Run existing tests, make sure there are no fails due to unresolved import, i.e. 
current import in `slowmo_comm.py` `from torch.distributed.algorithms._comm_hooks import allreduce_hook as default` 
fails, because there is no `allreduce_hook` in a master branch of PyTorch ( and in nightly version)

**Check list:**
- [ ] Was this **discussed and approved** via a GitHub issue? (not for typos or docs)
- [x] Did you read the [contributor guideline](https://github.com/pytorch/torchdistx/blob/main/CONTRIBUTING.md)?
- [x] Did you make sure that your **PR does only one thing** instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**?
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you **update the [CHANGELOG](https://github.com/pytorch/torchdistx/blob/main/CHANGELOG.md)**? (not for typos, docs, or minor internal changes)

### CODE REVIEW

File: src/python/torchdistx/slowmo/slowmo_optimizer.py [hunk 1]
```diff
@@ -174,11 +174,8 @@ def load_state_dict(self, state_dict):
         self.slowmo_lr = state_dict.pop("slowmo_lr")
         self.averager.step = state_dict.pop("step")
         self._base_optim.load_state_dict(state_dict)
-        # check that base optimizer's learning rate is stored in param_groups
-        if not (self.param_groups and self.param_groups[0]["lr"]):
-            raise ValueError(
-                "Base optimizer does not have parameters or learning rate specified."
-            )
+        if not self.param_groups:
```
Comment (rohan-varma) [inline src/python/torchdistx/slowmo/slowmo_optimizer.py:177]: why do we remove the learning rate check here?

**Review by rohan-varma — DISMISSED** [review]
LGTM, thanks!

File: src/python/torchdistx/slowmo/slowmo_optimizer.py (hunk 1 above)
Comment (aovladi) [reply src/python/torchdistx/slowmo/slowmo_optimizer.py:183]: @rohan-varma  I've added proper checks in this version, i.e. since it is possible to have multiple parameter groups and each of them should have `lr`, every group should have a check. Before, only first parameter group had a check

**Review by rohan-varma — APPROVED** [review]
